### PR TITLE
File Permission and Secrets Management

### PR DESCRIPTION
**Pull request by atotala**

### CODE REVIEW

File: hpedockerplugin/volume_manager.py [hunk 1]
```diff
@@ -1058,6 +1078,16 @@ def mount_volume(self, volname, vol_mount, mount_id):
         else:
             mount_dir = ''
 
+        if vol['fsOwner'] is not None:
```
Comment (wdurairaj) [inline hpedockerplugin/volume_manager.py]: you can add if condition like this
`if 'fsOwner' in vol and vol['fsOwner'] `

File: hpedockerplugin/volume_manager.py [hunk 2]
```diff
@@ -1058,6 +1078,16 @@ def mount_volume(self, volname, vol_mount, mount_id):
...
+            fs_owner = vol['fsOwner'].split(":")
+            uid = int(fs_owner[0])
+            gid = int(fs_owner[1])
+            os.chown(mount_dir, uid, gid)
+
+        if vol['fsMode'] is not None:
```
Comment (wdurairaj) [inline hpedockerplugin/volume_manager.py]: Same as above comment

File: hpedockerplugin/volume_manager.py [hunk 3]
```diff
@@ -1354,3 +1384,37 @@ def _get_required_qos_field(qos_detail):
         qos_filter['vvset_name'] = qos_detail['name']
 
         return qos_filter
+
+    def _decrypt_password(self, hpepluginconfig):
```
Comment (wdurairaj) [inline hpedockerplugin/volume_manager.py]: Will this method definition work correctly in a case, where the plugin is upgraded from a previous release, and now the _decrypt_password() is called on a plain text ?

File: hpedockerplugin/volume_manager.py (hunk 1 above)
Comment (atotala) [reply hpedockerplugin/volume_manager.py]: Made the change as per suggestion

File: hpedockerplugin/volume_manager.py (hunk 2 above)
Comment (atotala) [reply hpedockerplugin/volume_manager.py]: Made the change as per suggestion

File: hpedockerplugin/volume_manager.py (hunk 3 above)
Comment (atotala) [reply hpedockerplugin/volume_manager.py]: Made the change in function to work with previous release as well

File: hpedockerplugin/etcdutil.py [hunk 4]
```diff
@@ -179,3 +179,6 @@ def get_path_info_from_vol(self, vol):
             if 'path_info' in info and info['path_info'] is not None:
                 return json.loads(info['path_info'])
         return None
+
+    def read(self, key):
```
Comment (nilangekarss) [inline hpedockerplugin/etcdutil.py]: Can we rename def to read_etcd_key

File: hpedockerplugin/etcdutil.py [hunk 5]
```diff
@@ -179,3 +179,6 @@ def get_path_info_from_vol(self, vol):
             if 'path_info' in info and info['path_info'] is not None:
                 return json.loads(info['path_info'])
         return None
+
+    def read(self, key):
+        return self.client.read(key)
```
Comment (nilangekarss) [inline hpedockerplugin/etcdutil.py]: use try and except block

File: test/hpe_docker_unit_test.py [hunk 6]
```diff
@@ -80,8 +80,8 @@ def _mock_execute_api(self, mock_objects, plugin_api=''):
             String containing VolumePlugin API name
         :return: Nothing
         """
-        # import pdb
-        # pdb.set_trace()
+        #import pdb
+        #pdb.set_trace()
```
Comment (nilangekarss) [inline test/hpe_docker_unit_test.py]: check for pep8 errors and fix

File: docs/usage.md [hunk 7]
```diff
@@ -12,6 +12,9 @@ There are several optional parameters that can be used during volume creation:
 - size -- specifies the desired size in GB of the volume.
 - provisioning -- specifies the type of provisioning to use (thin, full, dedup).
 - flash-cache -- specifies whether flash cache should be used or not (True, False).
+- fsMode -- Represent the file mode to be applied to the root directory of the filesystem, in the form of octall digits.
```
Comment (nilangekarss) [inline docs/usage.md]: Update this in create_help.txt as well

**Pull request merged**